src/main/java/com/tabnine/general/StaticConfig.java

```diff
@@ -165,6 +168,24 @@ private static String getDistributionName() {
   }
 
   public static Path getBaseDirectory() {
+    // Unit tests don't initialize the application, so `ApplicationManager.getApplication` will
+    // return null.
+    Boolean isUnitTest =
+        ObjectUtils.doIfNotNull(ApplicationManager.getApplication(), Application::isUnitTestMode);
+    if (isUnitTest == null || isUnitTest) {
+      return getDefaultBaseDirectory();
+    }
+
+    String binariesFolderOverride = AppSettingsState.getInstance().getBinariesFolderOverride();
```

Comment:
does `AppSettingsState` is null in unit tests?

Reply:
Yes, because it is a registered service, and since the application is not initialized so are it's services

Reply:
so why do the unit tests test this function? 🤔

Reply:
They aren't. 
They use this function for the binary downloader tests and such, but before this pr, this function wasn't dependent on the application so it all worked fine.
Now that this function reads from the settings, it became dependent- so it needs a special handling